### PR TITLE
Advanced repo management

### PR DESCRIPTION
**Pull request by anton-sidelnikov**

### CODE REVIEW

File: orgs/opentelekomcloud-infra/repositories/test-to-be-deleted.yml [hunk 1]
```diff
@@ -12,33 +12,35 @@ test-to-be-deleted:
...
       required_pull_request_reviews:
         dismissal_restrictions:
-          users:
-            - anton-sidelnikov
-          teams:
-            - opentelekomcloud-infra/csm
+          users: []
+          teams: []
         dismiss_stale_reviews: true
         require_code_owner_reviews: false
```
Comment (lego963) [inline orgs/opentelekomcloud-infra/repositories/test-to-be-deleted.yml:39]: `require_code_owner_reviews: false` let's make it true

File: roles/gitcontrol/default/main.yml [hunk 2]
```diff
@@ -1,18 +1,6 @@
...
-github_repo_has_projects: true
-github_repo_has_wiki: true
-github_repo_private: false
-
 github_user: ''
 github_token: ''
-
-## Do nohing with Github by default.
-## Otherwise it's very easy to run into limits on the number of API calls.
 github_repos__state:
```
Comment (lego963) [inline roles/gitcontrol/default/main.yml:4]: these vars `github_repo*` has `__`. Is it ok?

**Review by lego963 — CHANGES_REQUESTED** [review]

File: orgs/opentelekomcloud-infra/repositories/gitstyring.yml [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+  delete_branch_on_merge: false
+  allow_merge_commit: false
+  allow_squash_merge: true
+  allow_rebase_merge : false
+  teams:
+    maintain:
+    pull:
+    push:
+    admin:
+      - csm
```
Comment (outcatcher) [inline orgs/opentelekomcloud-infra/repositories/gitstyring.yml:20]: Maybe `eco`?

File: playbooks/library/scan_organizations.py [hunk 4]
```diff
@@ -18,7 +18,7 @@ def main():
         for file in files:
             current_root = os.path.basename(root)
             a_yaml_file = open(os.path.join(root, file))
-            parsed_yaml_file = yaml.load(a_yaml_file)
+            parsed_yaml_file = yaml.load(a_yaml_file, Loader=yaml.BaseLoader)
```
Comment (outcatcher) [inline playbooks/library/scan_organizations.py:21]: Maybe use `SafeLoader` or `yaml.safe_load`?

File: roles/gitcontrol/files/github_api.py [hunk 5]
```diff
@@ -7,21 +7,39 @@
...
+bad_statuses = [400, 403, 404, 422]
 
 
 def read_yaml_file(path, org, endpoint, repo_name=None):
-    if endpoint == 'manage_collaborators':
+    if endpoint in ['manage_collaborators', 'branch_protection', 'options']:
         path += f'/{org}/repositories/{repo_name}.yml'
     with open(path, 'r') as file:
-        data = yaml.load(file)
+        data = yaml.load(file, Loader=yaml.FullLoader)
```
Comment (outcatcher) [inline roles/gitcontrol/files/github_api.py:21]: Why are you using `FullLoader`?

File: roles/gitcontrol/files/github_api.py [hunk 6]
```diff
@@ -7,21 +7,39 @@
...
+    if res.status_code in bad_statuses:
+        output += f'options change not applied: {res.status_code}, error is: {res.text}\n'
+    return print(output)
+
+
 def manage_collaborators(github_api, owner, repo_name, repo):
     output = ''
     collaborators = repo[repo_name]['collaborators']
+    teams = repo[repo_name]['teams']
     for clb in collaborators:
```
Comment (outcatcher) [inline roles/gitcontrol/files/github_api.py:43]: You can use `key, val in collaborators.items()` to iterate through both keys an values

File: roles/gitcontrol/files/github_api.py [hunk 7]
```diff
@@ -31,15 +49,37 @@ def manage_collaborators(github_api, owner, repo_name, repo):
                     headers=headers,
                     timeout=15)
                 if res.status_code in bad_statuses:
-                    output += f'user {user} not created: {res.status_code}'
-    return output
+                    output += f'user {user} not created: {res.status_code}, error is: {res.text}\n'
+    for team in teams:
```
Comment (outcatcher) [inline roles/gitcontrol/files/github_api.py:53]: Same as previous

File: roles/gitcontrol/files/github_api.py [hunk 8]
```diff
@@ -31,15 +49,37 @@ def manage_collaborators(github_api, owner, repo_name, repo):
...
+        if teams[team]:
```
Comment (outcatcher) [inline roles/gitcontrol/files/github_api.py:54]: You can use 
```python
if not teams[team]:
    continue
```
to decrease indentation :)

File: roles/gitcontrol/files/github_api.py [hunk 9]
```diff
@@ -31,15 +49,37 @@ def manage_collaborators(github_api, owner, repo_name, repo):
...
+                    timeout=15)
+                if res.status_code in bad_statuses:
+                    output += f'repo not added to team: {team}: {res.status_code}, error is: {res.text}\n'
+    return print(output)
+
+
+def update_branch_protection(github_api, owner, repo_name, repo):
+    output = ''
+    rules = repo[repo_name]['protection_rules']
+    branch_name = (list(rules.keys())[0])
```
Comment (outcatcher) [inline roles/gitcontrol/files/github_api.py:69]: Can be simmplified to `list(rules)[0]`

**Review by outcatcher — CHANGES_REQUESTED** [review]

**Comment by outcatcher** [comment]
recheck

File: README.md [hunk 10]
```diff
@@ -44,29 +43,21 @@ my_repo:
   allow_rebase_merge : false
   teams:
     maintain:
```
Comment (outcatcher) [inline README.md:45]: What does `maintain` permissions mean?

**Review by Polina-Gubina — APPROVED** [review]

**Review by lego963 — DISMISSED** [review]

**Review by lego963 — APPROVED** [review]

**Review by irina-pereiaslavskaia — APPROVED** [review]

**Review by outcatcher — APPROVED** [review]

**Pull request merged**